### PR TITLE
storing products map inside institutionParty

### PR DESCRIPTION
**Pull request by antonioT90**
When confirming the relationship of a manager, product's billingData and pricingPlan will be stored inside the InstitutionParty

Moreover, because we plan to clear the database, we are setting as required partyRegistry fields after added in the InstitutionParty model

ps: added zipCode to InstitutionUpdate

### CODE REVIEW

**Comment by antonioT90** [comment]
ASIS I wrote just the logic to store InstitutionProduct info inside the InstitutionParty entity.

Next I will write the logic to create InstitutionProduct info when the manager is confirmed and update the InstitutionParty.

**Comment by antonioT90** [comment]
added zipCode to InstitutionUpdate
added tests to check update on InstitutionParty

File: src/main/scala/it/pagopa/interop/partymanagement/api/impl/PublicApiServiceImpl.scala [hunk 1]
```diff
@@ -189,24 +189,47 @@ class PublicApiServiceImpl(
...
+  private def updateWithInstitutionProductInfo(
+    institutionParty: InstitutionParty,
+    relationship: PersistedPartyRelationship
+  ): InstitutionParty = {
+    relationship.billing.fold(institutionParty) { billing =>
+      val productId          = relationship.product.id
+      val institutionProduct = institutionParty.products
+        .find(_.product == productId)
+        .getOrElse(PersistedInstitutionProduct(product = productId, pricingPlan = None, billing = null))
+        .copy(pricingPlan = relationship.pricingPlan, billing = billing)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/partymanagement/api/impl/PublicApiServiceImpl.scala]: avoid using `null`s, even if it is temporary 🙏 
```suggestion
        .map(_.copy(pricingPlan = relationship.pricingPlan, billing = billing))
        .getOrElse(PersistedInstitutionProduct(product = productId, pricingPlan = relationship.pricingPlan, billing = billing))
```

File: src/main/protobuf/v1/relationship.proto [hunk 2]
```diff
@@ -51,7 +51,8 @@ message InstitutionUpdateV1 {
   optional string description = 2;
   optional string digitalAddress = 3;
   optional string address = 4;
-  optional string taxCode = 5;
+  optional string zipCode = 5;
+  optional string taxCode = 6;
```
Comment (beetlecrunch) [inline src/main/protobuf/v1/relationship.proto:55]: If you are going to wipe out all the persisted data, there is no problem, otherwise, be careful here.

Comment (antonioT90) [reply src/main/protobuf/v1/relationship.proto:55]: yes, we are going to wipe all the data

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**